runtime/crab/src/lib.rs

```diff
@@ -699,8 +699,14 @@ fn migrate() -> Weight {
 	// TODO: Move to S2S
 	// const CrabIssuingPalletId: PalletId = PalletId(*b"da/crais");
 
-	0
-	// RuntimeBlockWeights::get().max_block
+	for (key, value) in
+		migration::storage_iter::<(AccountId, Balance)>(b"Indices", b"Accounts").drain()
```

Comment:
How many key do we have in Crab approximately? Only the accounts which as assigned indices?

https://crab.subscan.io/event?address=&module=indices&event=indexassigned&startDate=&endDate=&startBlock=&endBlock=&timeType=date&version=1140

Reply:
Subscan found 3.

Reply:
@WoeOm Could you query the `IndicesAccounts` length?

Reply:
```
[
  [0x1a736d37504c2e3fb73dad160c55b2918ee7418a6531173d60d1f6a82d8f4d51754faa9acf0378f8c3543d9f132d85bc02000000, ["5HGTJmcw8hqFTQLCFwYQvusbga2SRT6FxiBjsgDxdLMLTyZk",1000000000,false]], 
  [0x1a736d37504c2e3fb73dad160c55b2918ee7418a6531173d60d1f6a82d8f4d5196262ec8dd0da5b6d99040f16d4a14f046090000, ["5HGTJmcw8hqFTQLCFwYQvusbga2SRT6FxiBjsgDxdLMLTyZk",1000000000,false]], 
  [0x1a736d37504c2e3fb73dad160c55b2918ee7418a6531173d60d1f6a82d8f4d51d82c12285b5d4551f88e8f6e7eb52b8101000000, ["5HgCRABJyoNTd1UsRwzErffZPDDfdYL3b1y3fZpG8hBScHC2",1000000000,false]]
]


https://apps.darwinia.network/#/js
console.log(await api.query.indices.accounts.entries());

```

Reply:
I think it's OK, will not have too many keys.

Reply:
> ```
> [
>   [0x1a736d37504c2e3fb73dad160c55b2918ee7418a6531173d60d1f6a82d8f4d51754faa9acf0378f8c3543d9f132d85bc02000000, ["5HGTJmcw8hqFTQLCFwYQvusbga2SRT6FxiBjsgDxdLMLTyZk",1000000000,false]], 
>   [0x1a736d37504c2e3fb73dad160c55b2918ee7418a6531173d60d1f6a82d8f4d5196262ec8dd0da5b6d99040f16d4a14f046090000, ["5HGTJmcw8hqFTQLCFwYQvusbga2SRT6FxiBjsgDxdLMLTyZk",1000000000,false]], 
>   [0x1a736d37504c2e3fb73dad160c55b2918ee7418a6531173d60d1f6a82d8f4d51d82c12285b5d4551f88e8f6e7eb52b8101000000, ["5HgCRABJyoNTd1UsRwzErffZPDDfdYL3b1y3fZpG8hBScHC2",1000000000,false]]
> ]
> 
> 
> https://apps.darwinia.network/#/js
> console.log(await api.query.indices.accounts.entries());
> ```

The js test shows that we didn't miss the migration?

Reply:
Not sure if they did some error handling.
But can not query index 1 from Polkadot Apps.

Reply:
Yes, but I can query from Darwinia Apps

Reply:
应该是 Darwinia Apps 解析 Types 时没有开启严格模式

Reply:
received 0xf83ee607164969887eaecab7e058ab3ba0f64c0cfe3f0b575fe45562cfc36bd500ca9a3b000000000000000000000000 (48 bytes), created 0xf83ee607164969887eaecab7e058ab3ba0f64c0cfe3f0b575fe45562cfc36bd500ca9a3b00000000000000000000000000 (49 bytes)

Looks like it will append 0 automactically.

Reply:
@WoeOm 能准确读一下 value的数据吗？ key是1，2 和2374，看看bytes length是 48还是49。

需要确定一下storage是否需要migrate.

Reply:
```
2021-09-17 09:49:49             DRR: Unable to decode storage indices.accounts: entry 2:: createType((AccountId,BalanceOf,bool)):: (AccountId,BalanceOf,Bool):: Decoded input doesn't match input, received 0xf83ee607164969887eaecab7e058ab3ba0f64c0cfe3f0b575fe45562cfc36bd500ca9a3b000000000000000000000000 (48 bytes), created 0xf83ee607164969887eaecab7e058ab3ba0f64c0cfe3f0b575fe45562cfc36bd500ca9a3b00000000000000000000000000 (49 bytes)
```

Reply:
从提示来看，正好是少了一个byte，用作bool.

Reply:
```
{
	"jsonrpc": "2.0",
	"result": [{
		"block": "0x39849e4de3a84c3d16f429705a1432921731fe8684c7ab054fc97a8e571495c8",
		"changes": [
			["0x1a736d37504c2e3fb73dad160c55b2918ee7418a6531173d60d1f6a82d8f4d51754faa9acf0378f8c3543d9f132d85bc02000000", "0xe62321ed84ec54791122f2ec72e9e36d3cb336ed358d6848a65b8410b405650a00ca9a3b00000000000000000000000000"],
			["0x1a736d37504c2e3fb73dad160c55b2918ee7418a6531173d60d1f6a82d8f4d5196262ec8dd0da5b6d99040f16d4a14f046090000", "0xe62321ed84ec54791122f2ec72e9e36d3cb336ed358d6848a65b8410b405650a00ca9a3b00000000000000000000000000"],
			["0x1a736d37504c2e3fb73dad160c55b2918ee7418a6531173d60d1f6a82d8f4d51d82c12285b5d4551f88e8f6e7eb52b8101000000", "0xf83ee607164969887eaecab7e058ab3ba0f64c0cfe3f0b575fe45562cfc36bd500ca9a3b000000000000000000000000"]
		]
	}],
	"id": 3
}

```

Reply:
看上去只有 5HgCRABJyoNTd1UsRwzErffZPDDfdYL3b1y3fZpG8hBScHC2 错误了

Reply:
Yes, there is one key (1) need migration.

Reply:
> ```
> {
> 	"jsonrpc": "2.0",
> 	"result": [{
> 		"block": "0x39849e4de3a84c3d16f429705a1432921731fe8684c7ab054fc97a8e571495c8",
> 		"changes": [
> 			["0x1a736d37504c2e3fb73dad160c55b2918ee7418a6531173d60d1f6a82d8f4d51754faa9acf0378f8c3543d9f132d85bc02000000", "0xe62321ed84ec54791122f2ec72e9e36d3cb336ed358d6848a65b8410b405650a00ca9a3b00000000000000000000000000"],
> 			["0x1a736d37504c2e3fb73dad160c55b2918ee7418a6531173d60d1f6a82d8f4d5196262ec8dd0da5b6d99040f16d4a14f046090000", "0xe62321ed84ec54791122f2ec72e9e36d3cb336ed358d6848a65b8410b405650a00ca9a3b00000000000000000000000000"],
> 			["0x1a736d37504c2e3fb73dad160c55b2918ee7418a6531173d60d1f6a82d8f4d51d82c12285b5d4551f88e8f6e7eb52b8101000000", "0xf83ee607164969887eaecab7e058ab3ba0f64c0cfe3f0b575fe45562cfc36bd500ca9a3b000000000000000000000000"]
> 		]
> 	}],
> 	"id": 3
> }
> ```

What's the rpc and parameters you are using?

Reply:
```
wss://crab-rpc.darwinia.network/

{"id":3,"jsonrpc":"2.0","method":"state_queryStorageAt","params":[["0x1a736d37504c2e3fb73dad160c55b2918ee7418a6531173d60d1f6a82d8f4d51754faa9acf0378f8c3543d9f132d85bc02000000","0x1a736d37504c2e3fb73dad160c55b2918ee7418a6531173d60d1f6a82d8f4d5196262ec8dd0da5b6d99040f16d4a14f046090000","0x1a736d37504c2e3fb73dad160c55b2918ee7418a6531173d60d1f6a82d8f4d51d82c12285b5d4551f88e8f6e7eb52b8101000000"]]}

```